parcels/particleset.py

```diff
@@ -71,23 +71,16 @@ class ParticleSet(object):
 
     :param grid: Grid object from which to sample velocity
     :param pclass: Optional class object that defines custom particle
-    :param lon: Optional list of initial longitude values for particles
-    :param lat: Optional list of initial latitude values for particles
-    :param start: Optional starting point for initilisation of particles
-                 on a straight line. Use start/finish instead of lat/lon.
-    :param finish: Optional end point for initilisation of particles on a
-                 straight line. Use start/finish instead of lat/lon.
-    :param start_field: Optional field for initialising particles stochastically
-                 according to the presented density field. Use instead of lat/lon.
-    :param size: Optional initial size of particle set, only required when using
-                 start/finish or start_field arguments
+    :param lon: List of initial longitude values for particles
+    :param lat: List of initial latitude values for particles
     """
 
-    def __init__(self, grid, pclass=JITParticle, lon=None, lat=None,
-                 start=None, finish=None, start_field=None, size=None):
+    def __init__(self, grid, pclass=JITParticle, lon=None, lat=None):
+        # Convert numpy arrays to one-dimensional lists
+        lon = lon.flatten() if isinstance(lon, np.ndarray) else lon
+        lat = lat.flatten() if isinstance(lat, np.ndarray) else lat
+        size = len(lon)
         self.grid = grid
```

Comment:
Ok, grid-switching at runtime indeed requires a lot of sanity checking and under-the-hood updates if we want to do this safely and is therefore not really supported at the moment. If you have a good use case for this functionality though, can you please file an issue, so that we can look into this separately? I don't think this impacts this PR.